scripts/install

```diff
@@ -109,70 +110,103 @@ ynh_add_fpm_config
 
 ynh_mysql_connect_as "$db_name" "$db_pwd" "$db_name" \
-    < "${final_path}/SQL/mysql.initial.sql"
+    < "$final_path/SQL/mysql.initial.sql"
 
 #=================================================
 # CONFIGURE ROUNDCUBE
 #=================================================
+ynh_print_info "Configuring roundcube..."
+
+rc_conf="$final_path/config/config.inc.php"
 
-rc_conf="${final_path}/config/config.inc.php"
 cp ../conf/config.inc.php "$rc_conf"
 
-ynh_replace_string "#DESKEY#" "$(ynh_string_random 24)" "$rc_conf"
-ynh_replace_string "#DBUSER#" "$db_name" "$rc_conf"
-ynh_replace_string "#DBPASS#" "$db_pwd" "$rc_conf"
-ynh_replace_string "#DBNAME#" "$db_name" "$rc_conf"
+ynh_replace_string "__DESKEY__" "$(ynh_string_random 24)" "$rc_conf"
+ynh_replace_string "__DBUSER__" $db_name "$rc_conf"
+ynh_replace_string "__DBPASS__" "$db_pwd" "$rc_conf"
+ynh_replace_string "__DBNAME__" $db_name "$rc_conf"
 
 #=================================================
 # INSTALL ADDITIONAL PLUGINS
 #=================================================
+ynh_print_info "Installing additional plugins..."
 
 # Create logs and temp directories
-mkdir -p "${final_path}/logs" "${final_path}/temp"
+mkdir -p "$final_path/"{logs,temp}
 
 # Install contextmenu and automatic_addressbook plugins
 # https://plugins.roundcube.net/packages/sblaisot/automatic_addressbook
 # https://plugins.roundcube.net/packages/johndoh/contextmenu
-exec_composer "$final_path" require \
-    "johndoh/contextmenu $contextmenu_version" \
-    "sblaisot/automatic_addressbook $automatic_addressbook_version"
+ynh_composer_exec --commands="require \
+    johndoh/contextmenu $contextmenu_version \
+    sblaisot/automatic_addressbook $automatic_addressbook_version"
+
 installed_plugins+=" 'contextmenu', 'automatic_addressbook',"
 
 # Install CardDAV plugin
-if [[ $with_carddav -eq 1 ]]; then
-  install_carddav "$final_path" \
-    && installed_plugins+=" 'carddav'," \
-    || echo "Unable to install CardDAV plugin" >&2
+if [ $with_carddav -eq 1 ]
+then
+    ynh_composer_exec --commands="require roundcube/carddav $carddav_version"
+
+    carddav_tmp_config="../conf/carddav.config.inc.php"
+
+    # Look for installed and supported CardDAV servers
+    for carddav_app in "owncloud" "baikal"
+    do
+        carddav_app_id=$(yunohost app list --installed -f $carddav_app \
+            --output-as json | grep -Po '"id":[ ]?"\K.*?(?=")' | head -1)
+        if [ -n "$carddav_app_id" ]
+        then
+            # Retrieve app settings and enable relevant preset
+            carddav_domain=$(ynh_app_setting_get $carddav_app_id domain)
+            carddav_path=$(ynh_app_setting_get $carddav_app_id path)
+            carddav_url="https://${carddav_domain}${carddav_path%/}"
+            ynh_replace_string "{${carddav_app}_url}" "$carddav_url" "$carddav_tmp_config"
+            sed -i "/\/\/\/\/ PRESET FOR: $carddav_app/,/\/\/\/\/ END: $carddav_app/s/^\/\///" "$carddav_tmp_config"
```

Comment:
Why not ynh_replace_string? (maybe because the string to replace is horrible, but I prefer asking)

Reply:
That a multi line sed, not a simple sed with substitution.

Reply:
Since config file is a php file, we can use a multi-lines comment like that:
```
/* PRESET FOR: owncloud
$prefs['ownCloud'] = array(
        // required attributes
        'name'         =>  'ownCloud',
        'username'     =>  '%u',
        'password'     =>  '%p',
        'url'          =>  '{owncloud_url}/remote.php/dav/addressbooks/users/%u/default/',

        // optional attributes
        'active'       =>  true,
        'readonly'     =>  false,
        'refresh_time' => '00:05:00',

        'fixed'        =>  array('username', 'password'),
        'hide'         =>  false,
);
// END: owncloud */
```

I have to test it too, but like that we could use `ynh_replace_string` instead of `sed`

Reply:
That would be better indeed